### DISCUSSION TITLE
Include AdGuard Annoyance Filters

### DISCUSSION THREAD

**Discussion by endocrine7132** [uBlockOrigin/uBOL-home#137] [Ideas]
I was wondering if including both the EasyList and AdGuard Annoyance filters would be possible as user-enabled filter options? Or if the size of these respective filters would be such that including both would bump uBOL up against the maximum MV3 filter list count?

I have found that AdGuard's versions of the Annoyance filters include some helpful cosmetic filtering rules that block pop-up overlays and other web annoyances on some of the sites I visit better than the EasyList variants do.

**Comment by endocrine7132** [comment]
I meant adding an option to enable the AdGuard variants to the currently included EasyList options. So for example, provide the option for a user to either enable EasyList Cookies or AdGuard Cookies list.

________________________________

**Reply by garry-ut99** [reply]
Continue here, because currently it has evolved into a discussion regarding "the whole existence of AdGuard lists in the whole uBlock extensions":
- https://github.com/uBlockOrigin/uBlock-issues/issues/3231

**Comment by gorhill** [comment]
Please report filtering issues through the _chat_ icon in the popup panel.

**Comment by Yuki2718** [comment]
Answer from @gorhill https://github.com/uBlockOrigin/uAssets/discussions/27718#discussioncomment-12656162

**Comment by Yuki2718** [comment]
@gorhill But is it possible to only add AdGuard Widget filter because there's no counerpart in EL/Fanboy family? An actual user complaint: `https://x.com/ref168622/status/1932997954362880505`.

**Reply by gorhill** [reply]
Why not just report what is missed by EasyList to EasyLIst?

**Reply by Yuki2718** [reply]
1. There's no dedicated list to address those Japanese-specific widgets in EL/FA, maybe can be added to FA though.
2. I don't have time or motivation to address them in EL/FA. Look, [L16-L1331 of the current Widget filter](https://github.com/AdguardTeam/AdguardFilters/blob/master/AnnoyancesFilter/Widgets/sections/widgets.txt#L16-L1331) are all for these blogroll widgets and the section is still growing.
3. There has been no single report for them in EL, making clear contrast with regular reports in AdguardFilters.

**Reply by gorhill** [reply]
My concern is overlap. If an AdGuard list is mostly not overlapping with ones we already have as stock list, I am open to add it, otherwise I prefer not. Each time I add a list which contains various mix of filters such as cosmetic, scriptlets, network, this causes many new files to be created and possibly many new injections of file on all sites, and this is something which worries me at the moment.

Currently I see 477 distinct scriptlet-related files in uBOL, and just for `example.com` domain, I see 35 distinct files injected as a result of scriptlet injections. With all five annoyance lists enabled, I see 60 distincts scriptlet injections, and that is for a site without any need for filtering. Hence why I really want the inclusion of new lists to be really justified, especially knowing that people have this tendency to enable as many lists as they can without a second thought.

So for me the question comes down to: is AdGuard Widgets mostly _not_ overlapping with existing lists? And if not, what would be a better name than "AdGuard Widgets" to convey purpose knowing that there is already "EasyList -- Social Widgets" and "EasyList -- Chat Widgets"?

**Reply by Yuki2718** [reply]
The list certainly overlaps with `EasyList -- Chat Widgets`. But it looks most of them currently are network rules and thus less of an issue. I don't know about future though, initially Fanboy Notification only addressed push notifications, but at some point it started to address mobile app banners, that's why I stopped using it to avoid redundancy with AdGuard Mobile app filter. Either list may start to address new classes of annoyances at some point in future. Regarding the name, maybe "AdGuard -- Annoying Widgets" as it addresses all sorts of widgets except for Social widgets?

**Reply by gorhill** [reply]
What if I merge it with existing list and rename to match combination of both lists? Just as is done for EasyList/uBO Cookies. Like "EasyList/AdGuard -- Other Widgets" -- could be merged with current "EasyList -- Chat Widgets"

**Reply by Yuki2718** [reply]
Looks fine, but `Other Widgets` will be somewhat strange. I think `Widgets` or `Annoying Widgets` will be better, or maybe there's some better name.
